code/processes/gateway.q

```diff
@@ -374,11 +375,20 @@ asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
    errstr:.gw.errorprefix,"only synchronous calls are allowed";
    [errstr:"";
    if[99h<>type servertype;
-     // its a list of servertypes e.g. `rdb`hdb
-     servertype:distinct servertype,();
-     errcheck:@[getserverids;servertype;{.gw.errorprefix,x}];
-     if[10h=type errcheck; errstr:errcheck];
-     queryattributes:()!();
+     if[all -11h = type each servertype;
+       // its a list of servertypes e.g. `rdb`hdb
+       servertype:distinct servertype,();
+       errcheck:@[getserverids;servertype;{.gw.errorprefix,x}];
+       if[10h=type errcheck; errstr:errcheck];
+       queryattributes:()!();
+      ];
+     if[all -6h = type each raze servertype;
+       // its a (nested) list of serverid(s) e.g. 1 2 3 4i/(1i;3i;5 6i)
+       // servertype requires all to be of same type
+       servertype:(),/:key o;
+       queryattributes:enlist[`servertype]!enlist distinct value[o]$[`servertype in cols value o;`servertype;`procs];
+       query[2;1]:@[value o;`procs]!value o;
```

Comment:
mentioned above, but it seems like this is expecting a very specific query format - is there any guarantee that we will only reach this piece of code if the query is in that format?

also probably need some comments here, it's not very clear what this is doing or why

Reply:
This code is specifically used for the DA API only. We'll add comments for this and change the conditional to bring it more in line with the function prior to the DA API changes. This will also make it more robust for the code to only be run when the DA API is used and resort to TorQ v4 logic otherwise.